### PR TITLE
feat: tracker now can track groups

### PR DESCRIPTION
**Pull request by gujjs**

### CODE REVIEW

File: vsdkx/addon/tracking/groupprocessor.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from vsdkx.addon.tracking.processor import TrackerProcessor
+
+
+class GroupTrackerProcessor(TrackerProcessor):
+    """
+    Track group boxes
+    """
+    def post_process(self, addon_object: AddonObject) -> AddonObject:
+        addon_object.inference.extra["tracked_groups"], \
+        addon_object.shared["trackable_groups"] = \
```
Comment (gujjs) [inline vsdkx/addon/tracking/groupprocessor.py]: @nicolexirakia what should we call these two keys?

Comment (nicolexirakia) [reply vsdkx/addon/tracking/groupprocessor.py]: @gujjs I really don't remember what we agreed on, I thought you kept notes on it. Probably one is 'detected_groups' and the other one 'tracked_groups'? But not 100% sure. Check the inference method one more time and see what happens to each one of these guys.

File: vsdkx/addon/tracking/processor.py [hunk 2]
```diff
@@ -49,15 +65,10 @@ def post_process(self, addon_object: AddonObject) -> AddonObject:
         # centroids of bounding boxes
         last_updated = {}
 
+        objects, bounding_boxes = self._ct.update(boxes)
         # exit immediately if no people boxes found
-        if len(addon_object.inference.boxes) == 0:
-            self._ct.update(addon_object.inference.boxes)
-            addon_object.inference.extra["tracked_objects"] = event_counter
```
Comment (nicolexirakia) [inline vsdkx/addon/tracking/processor.py:55]: @gujjs 'tracked_objects' is the amount of tracked objects and 'trackable_object' are the objects that were tracked on the current frame.

File: vsdkx/addon/tracking/groupprocessor.py (hunk 1 above)
Comment (nicolexirakia) [reply vsdkx/addon/tracking/groupprocessor.py]: also check the title of the task here: https://gitlab.com/natix/cvison/base/-/issues/111

Comment (gujjs) [reply vsdkx/addon/tracking/groupprocessor.py]: I don't think we agreed on anything. Nothing happens to these guys, according to you, they were logged before and never used

Comment (gujjs) [reply vsdkx/addon/tracking/groupprocessor.py]: that is whole another task that aims to rename existing used in production keys for the sake of better naming

**Comment by gujjs** [comment]
@nicolexirakia I think this is ready for a merge

File: vsdkx/addon/tracking/processor.py [hunk 3]
```diff
@@ -27,20 +27,36 @@ def __init__(self, addon_config: dict, model_settings: dict,
...
+            boxes (np.ndarray): bounding boxes of objects
         Returns:
-            (AddonObject): addon object has updated information for inference
-            result and/or shared information:
-                event_counter (int | dict): Amount of newly tracked events,
-                returns an int on unidirectional mode and a dict on a
-                bidirectional mode last_updated (dict): Filtered list of
-                trackable objects that were updated on the last frame
+            event_counter (int | dict): Amount of newly tracked events,
+            returns an int on unidirectional mode and a dict on a
```
Comment (nicolexirakia) [inline vsdkx/addon/tracking/processor.py]: this sentence is incomplete

File: vsdkx/addon/tracking/processor.py (hunk 2 above)
Comment (gujjs) [reply vsdkx/addon/tracking/processor.py:55]: I don't get it. is something wrong here?

Comment (nicolexirakia) [reply vsdkx/addon/tracking/processor.py:55]: no just ignore it, it's connected to the previous comment about renaming these values that's associated to another task.

File: vsdkx/addon/tracking/processor.py (hunk 3 above)
Comment (gujjs) [reply vsdkx/addon/tracking/processor.py]: edited

**Review by nicolexirakia — APPROVED** [review]

File: vsdkx/addon/tracking/groupprocessor.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from vsdkx.core.structs import AddonObject
+
+from vsdkx.addon.tracking.processor import TrackerProcessor
+
+
+class GroupTrackerProcessor(TrackerProcessor):
+    """
```
Comment (nicolexirakia) [inline vsdkx/addon/tracking/groupprocessor.py:7]: Mention the `TrackerProcessor` inheritance on the docstring.

**Pull request merged**